Writerside/topics/implementation/public-transit-service.md

```diff
@@ -1,9 +1,166 @@
+sub-routes". These sub-routes group trips that share identical stop sequences, ensuring compatibility with RAPTOR’s
+routing model. During post-processing, when the routing results are returned to the user, the sub-routes are converted
+back into the original GTFS routes to maintain user-friendly responses (TODO: Where is this happening? in the RAPTOR or
+in the service?).
+
+### Transfer Generation
+
+In GTFS, the transfer file is optional, meaning not all transfers between stops may be defined. To handle this, we
+introduced an optional transfer generation step in the `GtfsToRaptorConverter`. This step uses
+the `WalkTransferGenerator` to generate walking transfers between stops that are within a reasonable walking distance
+but are missing from the GTFS transfer data. Even when GTFS transfer data is provided, the generator can add missing,
+but viable, transfers and thus improve the connection results.
+
+The walking distances between stops are calculated using a `WalkCalculator`. This implementation uses a **beeline
+distance factor**, typically set to 1.3 based on literature [13, p. 43], to account for real-world walking conditions
+such as obstacles, detours, or building access routes. The calculator multiplies the straight-line distance by this
+factor to approximate a more realistic walking time. The `WalkCalculator` interface allows for easy replacement with
+more advanced footpath routing solutions, such as A* or A+ Landmark algorithms, if needed.
+
+### Handling Coordinates
+
+In routing requests, users can specify coordinates (latitude/longitude) or stops as the origin or destination. When
+coordinates are provided, the `GtfsRaptorService` first identifies the nearest stops using an existing KDTree that
+indexes all the stops. The walking time to these stops is calculated using the `WalkCalculator` instance, and only stops
+within a reasonable walking duration, defined in the service configuration at startup, are considered in the routing
+request.
+
+Once the RAPTOR routing engine processes the request, it calculates routes between the identified stops and returns the
+connection results. The `GtfsRaptorService` then enriches these results with walking segments (first mile and last
+mile), which represent the walk from the start coordinate to the selected origin stop and from the destination stop to
+the final coordinate.
+
+In summary, the `ConnectionRoutingService` performs pre-processing to identify nearby stops, routes the request using
+the RAPTOR engine, and post-processes the results by adding walking segments and re-aggregating sub-routes to provide
+user-friendly routing information.
+
+**Note:** Since we also support routing based on the latest departure, which involves routing backwards in time, we use
+the terms "source" and "target" stop in the code. However, in this explanation, we use the more traditional terms "
+origin" and "destination" for clarity.
```

Comment:
unnecessary white space between " and origin

Reply:
https://github.com/naviqore/documentation/pull/30/commits/5a613a8cbbb969087380d864b80035f745544b9d